src/creatures/players/player.cpp

```diff
@@ -7068,6 +7077,766 @@ void Player::dismount() {
 	defaultOutfit.lookMount = 0;
 }
 
+// Wings
+
+uint8_t Player::getLastWing() const {
+	const int32_t value = getStorageValue(PSTRG_WING_CURRENTWING);
+	if (value > 0) {
+		return value;
+	}
+	return static_cast<uint8_t>(kv()->get("last-wing")->get<int>());
+}
+
+uint8_t Player::getCurrentWing() const {
+	const int32_t value = getStorageValue(PSTRG_WING_CURRENTWING);
+	if (value > 0) {
+		return value;
+	}
+	return 0;
+}
+
+void Player::setCurrentWing(uint8_t wing) {
+	addStorageValue(PSTRG_WING_CURRENTWING, wing);
+}
+
+bool Player::hasAnyWing() const {
+	const auto &wings = g_game().attachedeffects->getWings();
+	return std::ranges::any_of(wings, [&](const auto &wing) {
+		return hasWing(wing);
+	});
+}
+
+uint8_t Player::getRandomWingId() const {
+	std::vector<uint8_t> availableWings;
+	const auto wings = g_game().attachedeffects->getWings();
+	for (const auto &wing : wings) {
+		if (hasWing(wing)) {
+			availableWings.emplace_back(wing->id);
+		}
+	}
+
+	const auto availableWingsSize = static_cast<int32_t>(availableWings.size() - 1);
```

Comment:
You need to validate this, same as here: https://github.com/opentibiabr/canary/pull/3236